src/Cardano/CoinSelection/LargestFirst.hs

```diff
@@ -20,77 +21,196 @@ import Cardano.Types
+--
+--      The /head/ of the list is always the unpaid output with the
+--      /greatest coin value/.
+--
+--      Entries are incrementally removed from the /head/ of the list as the
+--      algorithm proceeds, until the list is empty.
+--
+--  3.  An __/accumulated coin selection/__
+--
+--      This is initially /empty/.
+--
+--      Entries are incrementally added as each output is paid for, until the
+--      /unpaid output list/ is empty.
+--
+-- === Cardinality Rules
+--
+-- The algorithm requires that:
+--
+--  1.  Each output from the given /output list/ is paid for by /one or more/
+--      entries from the /initial UTxO set/.
+--
+--  2.  Each entry from the /initial UTxO set/ is used to pay for /at most one/
+--      output from the given /output list/.
+--
+--      (A single UTxO entry __cannot__ be used to pay for multiple outputs.)
+--
+-- === Order of Processing
+--
+-- The algorithm proceeds according to the following sequence of steps:
+--
+--  *   /Step 1/
+--
+--      Remove a single /unpaid output/ from the head of the
+--      /unpaid output list/.
+--
+--  *   /Step 2/
+--
+--      Repeatedly remove entries from the head of the /remaining UTxO list/
+--      until the total value of entries removed is /greater than or equal to/
+--      the value of the /unpaid output/.
+--
+--  *   /Step 3/
+--
+--      Use the /removed UTxO entries/ to pay for the /unpaid output/, by
+--      adding entries to the /accumulated coin selection/.
+--
+--  *   /Step 4/
+--
+--      If the /total selected value/ is greater than the value required for
+--      the current output, generate a /change output/ with the exact
+--      difference in value, and add it to the /accumulated coin selection/.
+--
+--  *   /Step 5/
+--
+--      If the /unpaid output list/ is empty, __terminate__ here.
+--
+--      Otherwise, return to /Step 1/.
+--
+-- === Successful Termination
+--
+-- The algorithm terminates __successfully__ if the /remaining UTxO list/ is
+-- not depleted before the /unpaid output list/ can be fully depleted (i.e., if
+-- all the outputs have been paid for).
+--
+-- The /accumulated coin selection/ and /remaining UTxO list/ are returned to
+-- the caller.
+--
+-- === Unsuccessful Termination
+--
+-- The algorithm terminates with an __error__ if:
+--
+--  1.  The /total value/ of the initial UTxO set (the amount of money
+--      /available/) is /less than/ the total value of the output list (the
+--      amount of money /required/).
+--
+--      See: __'ErrUtxoBalanceInsufficient'__.
+--
+--  2.  The /number/ of entries in the initial UTxO set is /smaller than/ the
+--      number of requested outputs.
+--
+--      Due to the nature of the algorithm, /at least one/ UTxO entry is
+--      required /for each/ output.
+--
+--      See: __'ErrUtxoNotFragmentedEnough'__.
+--
+--  3.  Due to the particular /distribution/ of values within the initial UTxO
+--      set, the algorithm depletes all entries from the set /before/ it is
```

Comment:
```suggestion
--      set, the algorithm depletes all entries from the UTxO set /before/ it is
```

Reply:
Fixed in https://github.com/input-output-hk/cardano-coin-selection/pull/7/commits/4f4e71bdfc57ddab0666b4dbb1d29b285fb0641f.